packages/react-virtualizer/src/hooks/useMeasureList.ts

```diff
@@ -124,11 +124,12 @@ export function useMeasureList<
 
         // cleanup previous container
         const prevEl = refArray.current[index];
-        if (prevEl) {
+        refArray.current[index] = undefined;
+        if (prevEl && !refArray.current.includes(prevEl)) {
```

Comment:
Sometimes items move position, but are still present - this prevents the RO from unobserving the previous element if it moved to another RO tracking index